### ISSUE TITLE
Package authors can use #if to perform version checks

### ISSUE THREAD

**Issue by terrajobst** [dotnet/sdk#14540]
[Spec](https://github.com/dotnet/designs/pull/164)

**Comment by marcpopMSFT** [comment]
Wait to implement until https://github.com/dotnet/designs/pull/165 has been decided on or not.

**Comment by terrajobst** [comment]
Based on community feedback I decided to abandon that PR and leave it as `OR_GREATER`. Ready to go!

**Comment by sfoslund** [comment]
Fixed by #14798, merged into 5.0.2xx

**Comment by chucker** [comment]
I want to conditionally implement `IAsyncDisposable` when available. Am I correct to assume that this is the right way of going about it?

```csharp
#if NETSTANDARD2_1 || NETCOREAPP3_0 || NETCOREAPP3_1 || NET5_0 || NETCOREAPP3_0_OR_GREATER
    public partial class MyClass : IAsyncDisposable
    {
        public async ValueTask DisposeAsync()
        {
            // impl
        }
    }
#endif
```

My thinking on this is that .NET Core 3.0 and 3.1 and .NET 5 didn't actually ship with this change, so people may have an SDK that doesn't support it.
